monetio/models/_rrfs_sd_mm.py

```diff
@@ -0,0 +1,166 @@
+    dset["latitude"] = dset["latitude"].isel(time=0)
+    dset["longitude"] = dset["longitude"].isel(time=0)
+    dset = dset.reset_coords()
+    dset = dset.set_coords(["latitude", "longitude"])
+
+    # These sums and units are quite expensive and memory intensive,
+    # so add option to shrink dataset to just surface when needed
+    if surf_only:
+        dset = dset.isel(z=0).expand_dims("z", axis=1)
+
+    
+
+    # convert "ug/kg to ug/m3"
+    for i in dset.variables:
+        if "units" in dset[i].attrs:
+            if "ug/kg" in dset[i].attrs["units"]:
+                # ug/kg -> ug/m3 using dry air density
+                dset[i] = dset[i] * dset["pres_pa_mid"] / dset["temperature_k"] / 287.05535
+                dset[i].attrs["units"] = r"$\mu g m^{-3}$"
+
+    
+    # Change the times to pandas format
+    dset["time"] = dset.indexes["time"].to_datetimeindex(unsafe=True)
+    # Turn off warning for now. This is just because the model is in julian time
+
+    return dset
+
+def _calc_hgt(f):
+    """Calculates the geopotential height in m from the variables hgtsfc and
+    delz. Note: To use this function the delz value needs to go from surface
+    to top of atmosphere in vertical. Because we are adding the height of
+    each grid box these are really grid top values
+
+    Parameters
+    ----------
+    f : xarray.Dataset
+        RRFS-CMAQ model data
+
+    Returns
+    -------
+    xr.DataArray
+        Geoptential height with attributes.
+    """
+    sfc = f.surfalt_m.load()
+    dz = f.dz_m.load() * -1.0
+    # These are negative in RRFS-CMAQ, but you resorted and are adding from the surface,
+    # so make them positive.
+    dz[:, 0, :, :] = dz[:, 0, :, :] + sfc  # Add the surface altitude to the first model level only
+    z = dz.rolling(z=len(f.z), min_periods=1).sum()
+    z.name = "alt_msl_m_full"
+    z.attrs["long_name"] = "Altitude MSL Full Layer in Meters"
+    z.attrs["units"] = "m"
+    return z
+
+
+def _calc_pressure(dset, surf_only=False):
+    """Calculate the mid-layer pressure in Pa from surface pressure
+    and ak and bk constants.
+
+    Interface pressures are calculated by:
+    phalf(k) = a(k) + surfpres * b(k)
+
+    Mid layer pressures are calculated by:
+    pfull(k) = (phalf(k+1)-phalf(k))/log(phalf(k+1)/phalf(k))
+
+    Parameters
+    ----------
+    dset : xarray.Dataset
+        RRFS-CMAQ model data
+
+    Returns
+    -------
+    xarray.DataArray
+        Mid-layer pressure with attributes.
+    """
+    p = dset.dp_pa.copy().load()  # Have to load into memory here so can assign levels.
+    srfpres = dset.surfpres_pa.copy().load()
+    for k in range(len(dset.z)):
+        if surf_only:
+           pres_2 = 0.0 + srfpres * 0.9978736
+           pres_1 = 0.0 + srfpres * 1.0
+        else:
+           pres_2 = dset.ak[k + 1] + srfpres * dset.bk[k + 1]
+           pres_1 = dset.ak[k] + srfpres * dset.bk[k]
+        p[:, k, :, :] = (pres_2 - pres_1) / np.log(pres_2 / pres_1)
```

Comment:
Maybe something like this
```suggestion
    surfpres = dset.surfpres_pa
    a = dset.ak
    b = dset.bk
    if surf_only:
        phalf_kp1 = 0.0 + surfpres * 0.9978736
        phalf_k = 0.0 + surfpres * 1.0
    else:
        phalf_kp1 = a.shift(z=-1) + surfpres * b.shift(z=-1)
        phalf_k = a + surfpres * b
    p = (phalf_kp1 - phalf_k) / np.log(phalf_kp1 / phalf_k)
```
going for consistency with the docstring variable names

Reply:
Since `ak` and `bk` are dataset attrs, not variables
```python
phalf_kp1 = a[1:] + surfpres * b[1:]
```